packages/kbn-coloring/src/palettes/utils.ts

```diff
@@ -0,0 +1,191 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import {
+  PaletteContinuity,
+  PaletteRegistry,
+  CustomPaletteParams,
+  DataBounds,
+  ColorStop,
+} from '../palettes/types';
+
+import {
+  DEFAULT_COLOR_STEPS,
+  DEFAULT_PALETTE_NAME,
+  DEFAULT_MAX_STOP,
+  DEFAULT_MIN_STOP,
+} from './constants';
+
+/** @internal **/
+export function calculateStop(
+  stopValue: number,
+  newMin: number,
+  oldMin: number,
+  oldInterval: number,
+  newInterval: number
+) {
+  if (oldInterval === 0) {
+    return newInterval + newMin;
+  }
+  return roundValue(newMin + ((stopValue - oldMin) * newInterval) / oldInterval);
+}
+
+/**
+ * This is a generic function to compute stops from the current parameters.
+ */
+export function getPaletteStops(
```

Comment:
It seems like a bunch of these helpers are duplicated as part of the heatmap renderer plugin - does it make sense to delete these and make it use the kbn/coloring counterparts? https://github.com/elastic/kibana/blob/fea4d2acfbc789cdbde56eb70a4a864ddd714e38/src/plugins/chart_expressions/expression_heatmap/public/components/helpers.ts#L22

Reply:
@flash1293  it was already done, but then we decided to split that work and do it in separate steps.

Reply:
Sounds good!